src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1638,8 +1641,9 @@ private unsafe void Initialize(Guid eventSourceGuid, string eventSourceName, str
                 m_eventPipeProvider = eventPipeProvider;
 #endif
                 Debug.Assert(!m_eventSourceEnabled);     // We can't be enabled until we are completely initted.
-                // We are logically completely initialized at this point.
-                m_completelyInited = true;
+                // We are logically completely initialized at this point, but set m_completelyInited after
```

Comment:
Removed the comments regarding "completely initialized" to reduce confusion, and since there is a null check for `m_etwProvider` in `DoCommand`, updated the comment there